### PR TITLE
Use strict mode whenever possible.

### PR DESCRIPTION
**Pull request by okonomiyaki3000**
Pull Request for Issue # .
### Summary of Changes

Javscript functions are set to work in strict mode. 
JQuery 'ready' function changed to concise format.
Minify some files that hadn't been minified before.
Other minor cleanup.
### Testing Instructions

These files are used by many admin pages including the global config, various listing and edit pages, permission forms, etc. A few of these files are only used by the `hathor` template. 

Most are included via some `JHtml` function. You can usually find which function by searching for the js filename. You can then find where that is used by searching joomla for 'classSuffix.functionName'. 
### Documentation Changes Required

Nope but it would be great if all future javascript development was required to be in strict mode. Also, if we had some guidelines on minification, that would be great too.

### CODE REVIEW

**Comment by okonomiyaki3000** [comment]
I believe this is a bad test. It requires that `sendTestMail` be defined on the global scope but there is no reason for that. This is a function that is used only as a handler for a single `click` event. Why make it global?

**Comment by okonomiyaki3000** [comment]
But anyway, I'll change it back for now.

**Comment by zero-24** [comment]
@yvesh can you take a look into the failing Drone Test?

**Comment by yvesh** [comment]
@zero-24 @okonomiyaki3000 when we want to use the strict mode, the javascript tests need to be adjusted to that. This should be done **before** merging that PR. 

@okonomiyaki3000 Can you look into fixing the JavaScript tests (we have a documentation on them at docs.joomla.org too)? Ping me when you need any help with that.

**Comment by okonomiyaki3000** [comment]
@yvesh I'm not familiar with drone tests and I don't know how to run them locally. Could you point me to more info about that? I couldn't find it on docs.joomla.org.

**Comment by dgt41** [comment]
Try https://docs.joomla.org/Running_JavaScript_Tests_for_the_Joomla_CMS

File: media/system/js/sendtestmail-uncompressed.js [hunk 1]
```diff
@@ -3,13 +3,8 @@
...
  */
-
-/**
- * Calls the sending process of the config class
- */
-var sendTestMail = function ()
-{
-	$ = jQuery;
+var sendTestMail = function () {
+	'use strict';
```
Comment (dgt41) [inline media/system/js/sendtestmail-uncompressed.js]: `$ = jQuery;`  adding this should fix the tests

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

File: media/system/js/sendtestmail-uncompressed.js (hunk 1 above)
Comment (okonomiyaki3000) [inline media/system/js/sendtestmail-uncompressed.js]: Thanks. It seems I forgot to put this in when I took the function back out of the ready function. Isn't it silly that the tests require this to be in the global scope though? It's kind of nonsense to me.

**Comment by okonomiyaki3000** [comment]
So there were basically two problems, one as @dgt41 mentioned and another which required improving the test. The `sendPermissions` function was relying on `this` to supply the id and value it needs. That's a bit risky in the best situations but here it was especially silly since `this` is a select in normal practice but it's `window` during the test. So I changed the test to work a little bit more like a real situation and used `event.target` instead `this`. No more errors.

**Comment by mbabker** [comment]
What's the status here?

**Comment by franz-wohlkoenig** [comment]
If this PR get no Response, it will be closed at 23th July 2017.

**Comment by ghost** [comment]
closed as stated above.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12604">issues.joomla.org/tracker/joomla-cms/12604</a>.</sub>

**Pull request closed without merging**